geth/node/rpc.go

```diff
@@ -94,6 +110,66 @@ func (c *RPCManager) Call(inputJSON string) string {
 	return c.makeJSONErrorResponse(ErrRPCServerTimeout)
 }
 
+// callNodeStream delivers giving request and body content to the external ethereum
+// (infura) RPCServer to process the request and returns response.
+func (c *RPCManager) callUpstreamStream(config *params.NodeConfig, body io.Reader) ([]byte, error) {
+	upstreamURL, err := url.Parse(config.UpstreamConfig.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	httpReq := httptest.NewRequest("POST", "/", body)
```

Comment:
But we create a new `Request` object here, right? And that SO question is about forwarding an incoming request to `http.Client`. I might be missing something, but
```go
req, err := http.NewRequest("POST", upstreamURL.String(), body)
```
should have `RequestURI` empty and should work just fine.